Investigate openmpi warnings

```
cd /home/ed/pio2/ParallelIO/build/src/clib && /usr/local/openmpi-2.0.2/bin/mpicc  -DCPRGNU -DHAVE_MPI -DINCLUDE_CMAKE_FCI -DLINUX -DLOGGING -DNETCDF_C_LOGGING_ENABLED -DTIMING -DUSE_PNETCDF_VARN -DUSE_PNETCDF_VARN_ON_READ -D_NETCDF -D_NETCDF4 -D_PNETCDF -I/home/ed/pio2/ParallelIO/src/clib -I/home/ed/pio2/ParallelIO/build/src/clib -I/usr/local/netcdf-4.4.1/include -I/usr/local/pnetcdf-1.8.1/include -I/home/ed/pio2/ParallelIO/src/gptl -I/home/ed/pio2/ParallelIO/build/src/gptl  -g -I/usr/local/pnetcdf-1.8.1/include -I/usr/local/netcdf-4.4.1/include -I/usr/local/openmpi-2.0.2/include    -std=c99 -o CMakeFiles/pioc.dir/pioc_support.c.o   -c /home/ed/pio2/ParallelIO/src/clib/pioc_support.c
/home/ed/pio2/ParallelIO/src/clib/pioc_support.c: In function ‘PIOc_read_nc_decomp’:
/home/ed/pio2/ParallelIO/src/clib/pioc_support.c:1087:18: warning: assignment makes integer from pointer without a cast [-Wint-conversion]
     mpi_type_int = mpi_type;
                  ^

```

```
/home/ed/pio2/ParallelIO/examples/c/example1.c: In function ‘main’:
/home/ed/pio2/ParallelIO/examples/c/example1.c:288:2: warning: ‘MPI_Errhandler_set’ is deprecated: MPI_Errhandler_set is superseded by MPI_Comm_set_errhandler in MPI-2.0 [-Wdeprecated-declarations]
  if ((ret = MPI_Errhandler_set(MPI_COMM_WORLD, MPI_ERRORS_RETURN)))
  ^~
In file included from /home/ed/pio2/ParallelIO/examples/c/example1.c:17:0:
/usr/local/openmpi-2.0.2/include/mpi.h:1352:20: note: declared here
 OMPI_DECLSPEC  int MPI_Errhandler_set(MPI_Comm comm, MPI_Errhandler errhandler)
                    ^~~~~~~~~~~~~~~~~~
[100%] Linking C executable example1

```

also:

`cd /home/ed/pio2/ParallelIO/build/src/clib && /usr/local/openmpi-2.0.2/bin/mpicc  -DCPRGNU -DHAVE_MPI -DINCLUDE_CMAKE_FCI -DLINUX -DLOGGING -DNETCDF_C_LOGGING_ENABLED -DTIMING -DUSE_PNETCDF_VARN -DUSE_PNETCDF_VARN_ON_READ -D_NETCDF -D_NETCDF4 -D_PNETCDF -I/home/ed/pio2/ParallelIO/src/clib -I/home/ed/pio2/ParallelIO/build/src/clib -I/usr/local/netcdf-4.4.1/include -I/usr/local/pnetcdf-1.8.1/include -I/home/ed/pio2/ParallelIO/src/gptl -I/home/ed/pio2/ParallelIO/build/src/gptl  -g -I/usr/local/pnetcdf-1.8.1/include -I/usr/local/netcdf-4.4.1/include -I/usr/local/openmpi-2.0.2/include    -std=c99 -o CMakeFiles/pioc.dir/pio_rearrange.c.o   -c /home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c
/home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c: In function ‘rearrange_comp2io’:
/home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c:860:21: warning: ‘MPI_Type_hvector’ is deprecated: MPI_Type_hvector is superseded by MPI_Type_create_hvector in MPI-2.0 [-Wdeprecated-declarations]
                     if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint) iodesc->llen * tsize, iodesc->rtype[i],
                     ^~
In file included from /home/ed/pio2/ParallelIO/src/clib/pio.h:16:0,
                 from /home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c:7:
/usr/local/openmpi-2.0.2/include/mpi.h:1803:20: note: declared here
 OMPI_DECLSPEC  int MPI_Type_hvector(int count, int blocklength, MPI_Aint stride,
                    ^~~~~~~~~~~~~~~~
/home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c:876:21: warning: ‘MPI_Type_hvector’ is deprecated: MPI_Type_hvector is superseded by MPI_Type_create_hvector in MPI-2.0 [-Wdeprecated-declarations]
                     if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint)iodesc->llen * tsize, iodesc->rtype[i],
                     ^~
In file included from /home/ed/pio2/ParallelIO/src/clib/pio.h:16:0,
                 from /home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c:7:
/usr/local/openmpi-2.0.2/include/mpi.h:1803:20: note: declared here
 OMPI_DECLSPEC  int MPI_Type_hvector(int count, int blocklength, MPI_Aint stride,
                    ^~~~~~~~~~~~~~~~
/home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c:904:13: warning: ‘MPI_Type_hvector’ is deprecated: MPI_Type_hvector is superseded by MPI_Type_create_hvector in MPI-2.0 [-Wdeprecated-declarations]
             if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint)iodesc->ndof * tsize, iodesc->stype[i],
             ^~
In file included from /home/ed/pio2/ParallelIO/src/clib/pio.h:16:0,
                 from /home/ed/pio2/ParallelIO/src/clib/pio_rearrange.c:7:
/usr/local/openmpi-2.0.2/include/mpi.h:1803:20: note: declared here
 OMPI_DECLSPEC  int MPI_Type_hvector(int count, int blocklength, MPI_Aint stride,
                    ^~~~~~~~~~~~~~~~
`

--- a/examples/c/example1.c
+++ b/examples/c/example1.c
@@ -285,7 +285,7 @@ int check_file(int ntasks, char *filename) {
 	/* Initialize MPI. */
 	if ((ret = MPI_Init(&argc, &argv)))
 	    MPIERR(ret);
-	if ((ret = MPI_Errhandler_set(MPI_COMM_WORLD, MPI_ERRORS_RETURN)))
+	if ((ret = MPI_Comm_set_errhandler(MPI_COMM_WORLD, MPI_ERRORS_RETURN)))
 	    MPIERR(ret);
 
 	/* Learn my rank and the total number of processors. */

--- a/src/clib/pio_rearrange.c
+++ b/src/clib/pio_rearrange.c
@@ -855,8 +855,8 @@ int rearrange_comp2io(iosystem_desc_t *ios, io_desc_t *iodesc, void *sbuf,
 
                     /*  The stride here is the length of the collected array (llen) */
 
-                    if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint) iodesc->llen * tsize, iodesc->rtype[i],
-                                                   recvtypes + i)))
+                    if ((mpierr = MPI_Type_create_hvector(nvars, 1, (MPI_Aint) iodesc->llen * tsize,
+                                                          iodesc->rtype[i], recvtypes + i)))
                         return check_mpi(NULL, mpierr, __FILE__, __LINE__);
 
                     if (recvtypes[i] == PIO_DATATYPE_NULL)
@@ -871,8 +871,8 @@ int rearrange_comp2io(iosystem_desc_t *ios, io_desc_t *iodesc, void *sbuf,
                 {
                     recvcounts[iodesc->rfrom[i]] = 1;
 
-                    if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint)iodesc->llen * tsize, iodesc->rtype[i],
-                                                   recvtypes + iodesc->rfrom[i])))
+                    if ((mpierr = MPI_Type_create_hvector(nvars, 1, (MPI_Aint)iodesc->llen * tsize, iodesc->rtype[i],
+                                                          recvtypes + iodesc->rfrom[i])))
                         return check_mpi(NULL, mpierr, __FILE__, __LINE__);
 
                     if (recvtypes[iodesc->rfrom[i]] == PIO_DATATYPE_NULL)
@@ -899,8 +899,8 @@ int rearrange_comp2io(iosystem_desc_t *ios, io_desc_t *iodesc, void *sbuf,
         if (scount[i] > 0 && sbuf != NULL)
         {
             sendcounts[io_comprank] = 1;
-            if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint)iodesc->ndof * tsize, iodesc->stype[i],
-                                           sendtypes + io_comprank)))
+            if ((mpierr = MPI_Type_create_hvector(nvars, 1, (MPI_Aint)iodesc->ndof * tsize,
+                                                  iodesc->stype[i], sendtypes + io_comprank)))
                 return check_mpi(NULL, mpierr, __FILE__, __LINE__);
 
             if (sendtypes[io_comprank] == PIO_DATATYPE_NULL)

